src/resources/views/crud/fields/select2.blade.php

```diff
@@ -1,76 +1,158 @@
 <!-- select2 -->
```

Comment:
Also on this topic. 

**Since the select2 fields that create items on-the-fly _make AJAX calls_, wouldn't it be better to use the ```select2_from_ajax``` field as a base for it?** It seems a bit sad that the relationship fields that allow you create items (huge feature!) would not allow you to use them on models with 1000+ entries (smaller but very important feature).

PROs:
- AJAX selects work well with big data sets (1000+ entries), non-AJAX selects don't; 
- you can use AJAX selects instead of non-AJAX; the reverse not so much;

CONs:
- AJAX selects are more difficult to set up at this time; but maybe we can fix the ```select2_from_ajax``` ease of use issue using Operations (see https://github.com/Laravel-Backpack/CRUD/issues/2175 ); then ```select2_from_ajax``` could become the go-to field for relationships, not ```select2```;

Or better yet, **if we create a separate ```select2_or_create```  field AND it is also usable on Models with a lot of entries, this could be a ```relationship``` field that renders all others obsolete/niche**. I think a single ```relationship``` field (see https://github.com/Laravel-Backpack/CRUD/issues/2050), that you'd use _all the time_, instead of picking different relationship fields depending on the relationship, is something we need to do next anyway.

In v3 and v4, relationship fields are kind of all over the place. They provide great ease of customization, anybody can do whatever they want. The fact that we have a diverse set of select fields will definitely be useful for different purposes in the future, to some. But maybe... there's an opportunity to create "_one relationship field to rule them all_"

![source-1](https://user-images.githubusercontent.com/1032474/70372616-29262100-18ea-11ea-8d37-67a7ae81ce26.gif)


I think there's an opportunity here _not only to add Create functionality_, but to _simplify how we're adding relationship fields_. I can't quite put my finger on it, and I'm not sure if my dream comes across clearly. But... Right now it looks to me like a good idea would be to:
- leave all select2 fields as they are right now; no extra features; ever;
- introduce the Create functionality in a new field (let's call it ```relationship``` for now);
- have this ```relationship``` field have all features needed out of a relationship field:
   - can be used on Models with a lot of entries (uses AJAX to fetch data);
   - can be used for multiple selections (just write ```'multiple' => true``` or ```'pivot' => true``` something like that);
   - allows the admin to Create entries on-the-fly (most likely using [Solution 2 here](https://github.com/Laravel-Backpack/CRUD/issues/2175#issuecomment-562835028));
   - (maybe in the future) allows the admin to reorder the related entries;

Alternatively, if we can't have a "relationship" field for ALL relationship types, we could have fields that are called exactly what the relationship method is: ```hasOne```, ```hasMany```, ```belongsToOne```, ```belongsToMany```. But I think it's possible to have just one.

Let's brainstorm this a little bit more, see what we come up with. **What do you think about having this one-size-fits-all field? Wouldn't it be great? Do you foresee any problems in having ```select2_from_ajax``` + on-the-fly + multiple in one field?**